src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java

```diff
@@ -2089,22 +2096,17 @@ private void paintCells(Graphics g, int rMin, int rMax, int cMin, int cMax) {
                 }
             }
         } else {
-            for(int row = rMin; row <= rMax; row++) {
+            for (int row = rMin; row <= rMax; row++) {
                 cellRect = table.getCellRect(row, cMin, false);
-                aColumn = cm.getColumn(cMin);
-                if (aColumn != draggedColumn) {
-                    columnWidth = aColumn.getWidth();
-                    cellRect.width = columnWidth - columnMargin;
-                    paintCell(g, cellRect, row, cMin);
-                }
-                for(int column = cMin+1; column <= cMax; column++) {
+                cellRect.x = getXPosition(cMax);
```

Comment:
Is this not a problem that `cellRect` is calculated for `cMin` but `cellRect.x` is calculated for `cMax` which means `cellRect.y` would be of `cMin` column but `cellRect.x` would be of `cMax` column. SImilarly for the `cellRect.width`...You probably can get away maybe because usually all cells have same width but I guess it will be a problem if different cells have different width...Please check..